City-of-Helsinki/parking-permits-admin-ui
# PV-346: Add announcements superadmin panel

## Description

This PR adds new views to the superadmin side of the UI for listing, viewing and creating announcements.

## Context

The superadmin needs to be able to send announcements to the customers through the admin UI. This feature allows the superadmin to do just that.

[PV-346](https://helsinkisolutionoffice.atlassian.net/browse/PV-346)
[Related back-end PR](https://github.com/City-of-Helsinki/parking-permits/pull/279)

## How Has This Been Tested?

Manual poking, in conjuction with the back-end implementation. No unit tests.

## Manual Testing Instructions for Reviewers

- Confirm that you can find the announcements under the admin page's side navigation bar
- Clicking the announcements section should take you to the announcement list view

### List view
- Should show a list of announcements
- If there aren't any announcements, should show an empty list
- Clicking an announcement on the list should take you to the announcement detail view
- Clicking the create announcement button should take you to the announcement creation view
- Click the create announcement button, it should take you to the creation view

### Creation view
- Should have subject and content fields for EN/FI/SV
- Should have parking zones as checkboxes
- The checkboxes should have a controller checkbox (select/deselect all)
- Should have submit and cancel buttons at the bottom of the page
- When clicked, the cancel button should take you back to the list view
- The submit button should be disabled initially
- The submit button should be enabled only when all of the fields have been filled
- The fields should show error messages if invalid (no checkboxes checked, no text given)
- After filling the form, it should create a new announcement and take you back to the list view with the new announcement in the list

### Detail view
- Should show all the information about the announcement
- Should have a button on the bottom of the page that takes the user back to the list view

## Screenshots

I'd add some screenshots if the authentication system hadn't suddenly stopped working.
